pkg/core/http.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaosdaemon/tproxyconfig"
+	"github.com/pingcap/errors"
+)
+
+const (
+	TargetRequest  tproxyconfig.PodHttpChaosTarget = "Request"
+	TargetResponse tproxyconfig.PodHttpChaosTarget = "Response"
+)
+
+const (
+	HTTPAbortAction  = "abort"
+	HTTPDelayAction  = "delay"
+	HTTPConfigAction = "config"
+)
+
+var _ AttackConfig = &HTTPAttackConfig{}
+
+type HTTPAttackConfig struct {
+	CommonAttackConfig
+	Config   tproxyconfig.Config
+	ProxyPID int
+
+	Logger logr.Logger
+}
+
+func (c HTTPAttackConfig) RecoverData() string {
+	data, _ := json.Marshal(c)
+	return string(data)
+}
+
+type HTTPAttackOption struct {
+	CommonAttackConfig
+	ProxyPorts []uint                            `json:"proxy_ports,omitempty"`
+	Rule       tproxyconfig.PodHttpChaosBaseRule `json:"rule"`
+	Path       string                            `json:"path"`
+}
+
+func NewHTTPAttackOption() *HTTPAttackOption {
+	port := int32(0)
+	path := ""
+	method := ""
+	code := int32(0)
+	return &HTTPAttackOption{
+		CommonAttackConfig: CommonAttackConfig{
+			Kind: HTTPAttack,
+		},
+		ProxyPorts: nil,
+		Rule: tproxyconfig.PodHttpChaosBaseRule{
+			Target: "",
+			Selector: tproxyconfig.PodHttpChaosSelector{
+				Port:   &port,
+				Path:   &path,
+				Method: &method,
+				Code:   &code,
+			},
+		},
+	}
+}
+
+func (o *HTTPAttackOption) PreProcess() (*HTTPAttackConfig, error) {
+	var c tproxyconfig.Config
+	zapLogger, err := zap.NewDevelopment()
+	if err != nil {
+		return nil, err
+	}
+	logger := zapr.NewLogger(zapLogger).WithName("HTTP Attack")
+	switch o.CommonAttackConfig.Action {
+	case HTTPAbortAction, HTTPDelayAction:
+		if *o.Rule.Selector.Path == "" {
```

Comment:
As mentioned above, we need more complex code to deal with the `PodHttpChaosBaseRule`.

Reply:
I agree with you.

Reply:
Fixed.